ui/packages/tidb-dashboard-lib/src/apps/Statement/pages/Detail/PlanBind.tsx

```diff
@@ -0,0 +1,303 @@
+import React, { useContext, useState, useMemo, useRef, useEffect } from 'react'
+import { Space, Button, Modal, Tooltip, Radio, Alert } from 'antd'
+import { InfoCircleOutlined } from '@ant-design/icons'
+
+import { useClientRequest } from '@lib/utils/useClientRequest'
+import { StatementModel } from '@lib/client'
+import { useTranslation } from 'react-i18next'
+import { IPageQuery } from '.'
+import { StatementContext } from '../../context'
+import { CardTable } from '@lib/components'
+import styles from './PlanBind.module.less'
+import { planColumns as genPlanColumns } from '../../utils/tableColumns'
+import {
+  SelectionMode,
+  CheckboxVisibility
+} from 'office-ui-fabric-react/lib/DetailsList'
+import { Selection } from 'office-ui-fabric-react/lib/Selection'
+
+interface PlanBindProps {
+  query: IPageQuery
+  plans: StatementModel[]
+}
+
+const PlanBind = ({ query, plans }: PlanBindProps) => {
+  const ctx = useContext(StatementContext)
+  const { t } = useTranslation()
+  const { data: planBindingStatus } = useClientRequest((reqConfig) =>
+    ctx!.ds.statementsPlanBindStatusGet!(
+      query.digest!,
+      query.beginTime!,
+      query.endTime!,
+      reqConfig
+    )
+  )
+
+  const [hasPlanToBind, setHasPlanToBind] = useState(false)
+  const [boundPlanDigest, setBoundPlanDigest] = useState<string | null>(null)
+  const [showPlanBindModal, setShowPlanBindModal] = useState(false)
+
+  const handleModalVisibility = (visibility: boolean) => {
```

Comment:
why do we need to wrap setState by handleXxx.

Reply:
You're right, it's unnecessary to wrap the setState.